pkg/controller/nstemplatetier/nstemplatetier_controller.go

```diff
@@ -190,23 +267,69 @@ func (r *ReconcileNSTemplateTier) activeTemplateUpdateRequests(logger logr.Logge
+	for _, item := range templateUpdateRequests.Items {
+		items[item.Name] = item.DeletionTimestamp
+	}
+	logger.Info("checking templateUpdateRequests", "items", items)
 	count := 0
 	for _, tur := range templateUpdateRequests.Items {
+		logger.Info("checking templateUpdateRequest", "name", tur.Name, "deleted", util.IsBeingDeleted(&tur))
 		if util.IsBeingDeleted(&tur) {
 			// ignore when already being deleted
+			logger.Info("skipping TemplateUpdateRequest as it is already being deleted", "name", tur.Name)
 			continue
 		}
+
+		// delete when in `complete=true` (reason=updated) or when in `complete=false/reason=failed` status conditions
 		if condition.IsTrue(tur.Status.Conditions, toolchainv1alpha1.TemplateUpdateRequestComplete) ||
 			condition.IsFalseWithReason(tur.Status.Conditions, toolchainv1alpha1.TemplateUpdateRequestComplete, toolchainv1alpha1.TemplateUpdateRequestUnableToUpdateReason) {
-			// delete when in `complete=true` (reason=updated) or when in `complete=false/reason=failed` status conditions
+			if err := r.incrementCounters(logger, tier, tur); err != nil {
+				return -1, false, err
+			}
 			if err := r.client.Delete(context.TODO(), &tur); err != nil {
+				if errors.IsNotFound(err) {
+					logger.Info("skipping failed TemplateUpdateRequest as it was already deleted", "name", tur.Name)
+					continue
+				}
 				return -1, false, errs.Wrapf(err, "unable to delete the TemplateUpdateRequest resource '%s'", tur.Name)
 			}
-			log.Info("deleted TemplateUpdateRequest", "name", tur.Name)
+			// will exit the reconcile loop
 			return -1, true, nil
 		}
 		count++
 	}
-	logger.Info("found active TemplateUpdateRequest for the current tier", "count", count)
+	logger.Info("found active TemplateUpdateRequests for the current tier", "count", count)
 	return count, false, nil
 }
+
+// incrementCounters looks-up the latest entry in the `status.updates` and increments the `Total` and `Failures` counters
+func (r *NSTemplateTierReconciler) incrementCounters(logger logr.Logger, tier *toolchainv1alpha1.NSTemplateTier, tur toolchainv1alpha1.TemplateUpdateRequest) error {
+	if len(tier.Status.Updates) == 0 {
+		return fmt.Errorf("no entry in the `Status.Updates`")
+	}
+	latest := tier.Status.Updates[len(tier.Status.Updates)-1]
+	if condition.IsFalseWithReason(tur.Status.Conditions, toolchainv1alpha1.TemplateUpdateRequestComplete, toolchainv1alpha1.TemplateUpdateRequestUnableToUpdateReason) {
+		latest.Failures++
+		latest.FailedAccounts = append(latest.FailedAccounts, tur.Name)
+	}
+	tier.Status.Updates[len(tier.Status.Updates)-1] = latest
+	if err := r.client.Status().Update(context.TODO(), tier); err != nil {
+		return err
+	}
+	logger.Info("incrementing counter after TemplateUpdateRequest completed", "name", tur.Name)
+	return nil
+}
+
+// markUpdateRecordAsCompleted looks-up the latest entry in the `status.updates` and sets the `CompletionTime` to `metav1.Now()`,
+// which means that the whole update process is done
```

Comment:
Or maybe there was nothing to update, in this case the ComplitionTime means the time when we last checked if there is something to update. Not sure if it's worth reflecting somewhere... Here or/and in API.
Just thinking loud :)

Reply:
ok, let me rephrase a bit the comment ;)
See bbfac63